src/components/Menu/Menu.tsx

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2023 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { FC, ReactNode, useMemo } from "react";
+import { Root, Trigger, Portal, Content } from "@radix-ui/react-dropdown-menu";
+import { FloatingMenu } from "./FloatingMenu";
+import { Drawer } from "vaul";
+import classnames from "classnames";
+import drawerMenu from "./DrawerMenu.module.css";
+import { MenuContext } from "./MenuContext";
+import { DrawerMenu } from "./DrawerMenu";
+import { useTouchscreen } from "../../utils/useTouchscreen";
+
+interface Props {
+  /**
+   * The menu title.
+   */
+  title: string;
+  /**
+   * Whether the menu is open.
+   */
+  open: boolean;
+  /**
+   * Event handler called when the open state of the menu changes.
+   */
+  onOpenChange: (open: boolean) => void;
+  /**
+   * The button that opens the menu. This must be a component that accepts a ref
+   * and spreads props.
+   * https://www.radix-ui.com/primitives/docs/guides/composition
+   */
+  trigger: ReactNode;
```

Comment:
Those three props seems slightly contradictory? 
I imagine that `trigger` is the main way to open the menu and `open` is here more for tests and storybook?